File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        j=1
        profit=0
        smallest=prices[0]
        while(j<len(prices)):
            pro=prices[j]-smallest
            if prices[j]<smallest:
                smallest=prices[j]
            if pro>profit:
                profit=pro
            j+=1
        return profit
        

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int m=prices[0];
        int ans=0;
        for(int i=1;i<prices.size();i++){
            ans=max(prices[i]-m,ans);
            m=min(m,prices[i]);
        }
        return ans;
    }
};